spec/yaml/swiftnav/sbp/ssr.yaml

```diff
@@ -47,19 +47,163 @@ definitions:
+    fields:
+    - time:
+        type: GPSTime
+        desc: GNSS time of the STEC data
+    - num_msgs:
+        type: u8
+        desc: Number of messages in the dataset
+    - seq_num:
+        type: u8
+        desc: Postion of this message in the dataset
+    - ssr_update_interval:
+        type: u16
+        desc: update interval in seconds
+    - iod_ssr:
+        type: u8
+        desc: range 0 - 15
+
+ - GriddedCorrectionHeader:
+    short_desc: Header for MSG_SSR_GRIDDED_CORRECTION
+    desc: |
+      The 3GPP message contains nested variable length arrays
+      which are not suppported in SBP, so each grid point will
+      be identified by the index.
+    fields:
+    - time:
+        type: GPSTime
+        desc: GNSS time of the STEC data
+    - num_msgs:
+        type: u8
+        desc: Number of messages in the dataset
+    - seq_num:
+        type: u8
+        desc: Postion of this message in the dataset
+    - ssr_update_interval:
+        type: u16
+        desc: update interval in seconds
+    - iod_ssr:
+        type: u8
+        desc: range 0 - 15
+    - tropo_quality:
+        type: u8
+        desc: troposphere quality indicator
+    - index:
+        type: u8
+        desc: index of the grid point
+
+ - STECSatElement:
+     short_desc: SSR code biases corrections for a particular satellite.
+     desc: |
+       Code biases are to be added to pseudorange.
+       The corrections are conform with typical RTCMv3 MT1059 and 1065.
+     fields:
+     - sv_id:
+         type: SvId
+         desc: Unique space vehicle identifier
+     - stec_quality_indicator:
+         type: u8
+         desc: quality of STEC data
+     - stec_coeff:
+         type: array
+         size: 4
+         fill: s16
+         desc: coefficents of the STEC polynomial
+
+ - TroposphericDelayCorrection:
+    short_desc: troposphere delay correction
+    desc: |
+      Contains wet vertical and hydrostatic vertical delay
+    fields:
+    - hydro:
+        type: s16
+        desc: hydrostatic vertical delay
+    - wet:
+        type: s8
+        desc: wet vertical delay
+
+ - STECResidual:
+    desc: STEC residual
+    fields:
+    - sv_id:
+        type: SvId
+        desc: space vehicle identifier
+    - residual:
+        type: s16
+        desc: STEC residual
+
+ - GridElement:
+    short_desc: Grid datum for troposphere and STEC residuals
+    desc: |
+      Contains one tropo datum, plus STEC residuals for each space
+      vehicle
+    fields:
+     - tropo_delay_correction:
+         type: TroposphericDelayCorrection
+         desc: Wet and Hydrostatic Vertical Delay
+     - STEC_residuals:
+         type: array
+         fill: STECResidual
+         desc: STEC Residual for the given space vehicle
+
+ - GridDefinitionHeader:
+    short_desc: Defines the grid for STEC and tropo grid messages
+    desc: |
+      Defines the grid for STEC and tropo grid messages.
+      Also includes an RLE encoded validity list.
+    fields:
```

Comment:
One shortcoming of the qualcomm grid vs ublox's is that it be nice to have a "grid ID" in the grid definition and correction messages. Then you could generate a series of smaller grids to cover a region and use the rover's position to filter out grids that are not relevant at the caster endpoint to reduce data rate.

Reply:
we could always implement both; should we?  @anth-cole @benjamin0 
if so, can you point me to the Ublox spec?

Reply:
Let's stick with the Qualcomm proposal for now to start simple and get things up and running then we can always rev this message later.